syncd/syncd_flex_counter.cpp

```diff
@@ -208,53 +219,58 @@ void FlexCounter::removePort(
     // Remove flex counter if counter IDs map is empty
     if (fc.m_queueCounterIdsMap.empty() && fc.m_portCounterIdsMap.empty() && fc.m_queueAttrIdsMap.empty())
     {
-        removeInstance(pollInterval);
+        removeInstance(instanceId);
     }
 }
 
 void FlexCounter::removeQueue(
         _In_ sai_object_id_t queueVid,
-        _In_ uint32_t pollInterval)
+        _In_ std::string instanceId)
 {
     SWSS_LOG_ENTER();
 
-    bool found = false;
-    FlexCounter &fc = getInstance(pollInterval);
+    FlexCounter &fc = getInstance(instanceId);
 
     auto counterIter = fc.m_queueCounterIdsMap.find(queueVid);
-    if (counterIter != fc.m_queueCounterIdsMap.end())
+    if (counterIter == fc.m_queueCounterIdsMap.end())
     {
-        fc.m_queueCounterIdsMap.erase(counterIter);
-        found = true;
+        SWSS_LOG_ERROR("Trying to remove nonexisting queue counter Ids 0x%lx", queueVid);
+        if (fc.m_queueCounterIdsMap.empty() && fc.m_portCounterIdsMap.empty() && fc.m_queueAttrIdsMap.empty())
```

Comment:
>fc.m_queueCounterIdsMap.empty() && fc.m_portCounterIdsMap.empty() && fc.m_queueAttrIdsMap.empty( [](start = 12, length = 96)

lots of duplicated code to check empty counter id, suggest to have a function to check it.

Reply:
for example 

IsCounterIdsMapEmpty()

---
In reply to: [166065518](https://github.com/Azure/sonic-sairedis/pull/289#discussion_r166065518) [](ancestors = 166065518)